### PR TITLE
Client logging fixes and improvements

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: clients/ts/signalr/src/Transports.ts [hunk 1]
```diff
@@ -313,17 +314,17 @@ export class LongPollingTransport implements ITransport {
...
-function getDataLength(data: any): number {
-    let length: number = null;
+function getDataDetail(data: any): string {
+    let length: string = null;
     if (data instanceof ArrayBuffer) {
-        length = data.byteLength;
-    } else if (data instanceof String) {
-        length = data.length;
+        length = `Binary data of length ${data.byteLength}`;
+    } else if (typeof data === "string") {
```
Comment (BrennanConroy) [inline clients/ts/signalr/src/Transports.ts]: Whats wrong with `instanceof String`?

File: clients/ts/FunctionalTests/ts/TestLogger.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+import { ConsoleLogger, ILogger, LogLevel } from "@aspnet/signalr";
 
 export class TestLogger implements ILogger {
     public static instance: TestLogger = new TestLogger();
+    private static consoleLogger: ConsoleLogger = new ConsoleLogger(LogLevel.Trace);
 
     public messages: Array<[LogLevel, string]> = [];
 
     public log(logLevel: LogLevel, message: string): void {
+        // Capture log message so it can reported later
```
Comment (BrennanConroy) [inline clients/ts/FunctionalTests/ts/TestLogger.ts]: can **be**

File: clients/ts/signalr/src/Transports.ts (hunk 1 above)
Comment (JamesNK) [reply clients/ts/signalr/src/Transports.ts]: `instanceof String` will match `new String("Hello World")` 🤷‍♂️ 

https://stackoverflow.com/questions/16792051/how-to-instanceof-a-primitive-string-string-literal-in-javascript

**Comment by JamesNK** [comment]
🆙 📅 

Added extra logging throughout TS client.

File: clients/ts/FunctionalTests/ts/TestLogger.ts [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+    private static consoleLogger: ConsoleLogger = new ConsoleLogger(LogLevel.Trace);
 
     public messages: Array<[LogLevel, string]> = [];
 
     public log(logLevel: LogLevel, message: string): void {
+        // Capture log message so it can be reported later
         this.messages.push([logLevel, message]);
+
+        // Also write to browser console
+        TestLogger.consoleLogger.log(logLevel, message);
```
Comment (anurse) [inline clients/ts/FunctionalTests/ts/TestLogger.ts]: Nice, thanks for catching that too!

File: clients/ts/signalr/src/Transports.ts (hunk 1 above)
Comment (anurse) [reply clients/ts/signalr/src/Transports.ts]: Yeah, `instanceof String` **only** matches `new String(...)`, not literals. So, `typeof` is the idomatic way to do this here

Comment (anurse) [reply clients/ts/signalr/src/Transports.ts]: image

Comment (anurse) [reply clients/ts/signalr/src/Transports.ts]: JavaScript is weird yo.

**Pull request merged**